src/target.cpp

```diff
@@ -1544,8 +1554,9 @@ ZigLLVM_EnvironmentType target_default_abi(ZigLLVM_ArchType arch, Os os) {
         case OsHurd:
             return ZigLLVM_GNU;
         case OsUefi:
-        case OsWindows:
             return ZigLLVM_MSVC;
+        case OsWindows:
+            return target_get_win32_abi();
```

Comment:
I'm ok with using cygwin pty to determine if the native ABI is mingw or not. But the function `target_default_abi` is not for native target - it's a pure function that returns the default ABI based on a given arch and os. So the logic that checks for a cygwin pty should be in the `get_native_target` function or called from there.